code/game/objects/structures/crates_lockers/closets/cardboardbox.dm

```diff
@@ -48,7 +48,7 @@
 			break
 		if(Snake)
 			alerted = viewers(7,src)
-	..()
+	. = ..()
```

Comment:
shouldn't this be checked if it returns false since the above part doesn't have COMSIG_CLOSET_PRE_OPEN checks

Reply:
I've now rewritten this proc. I utilise the cooldown defines, SECONDS define with a local var instead of a local #define for cooldown length, early returns when the box doesn't need to alert, better variable names and tested it.